packages/dashboard/src/components/tasks/task-alert.tsx

```diff
@@ -143,112 +145,112 @@ export function TaskAlertHandler({ alerts, removeAlert }: TaskAlertHandlerProps)
-                showAlertMessage += ` ${Math.round(ackSecondsAgo)}s ago`;
-              }
-              showAlert('success', showAlertMessage);
-            } else {
-              console.log(`Failed to acknowledge alert ID ${alert.task_id}`);
-              showAlert('error', `Failed to acknowledge alert ID ${alert.task_id}`);
-            }
-          })();
-        };
-
-        if (alert.acknowledgedBy) {
-          return (
-            <AlertDialog
-              key={alert.task_id}
-              onDismiss={dismissAlert}
-              acknowledgedBy={alert.acknowledgedBy}
-              title={alert.title}
-              progress={alert.progress}
-              alertContents={alert.content}
-              backgroundColor={alert.color}
-            />
-          );
+    (async () => {
+      const ackResponse = (
+        await rmf?.alertsApi.acknowledgeAlertAlertsAlertIdPost(taskAlert.task_id)
+      ).data;
+      if (ackResponse.id !== ackResponse.original_id) {
+        let showAlertMessage = `Alert ${ackResponse.original_id} acknowledged`;
+        if (ackResponse.acknowledged_by) {
+          showAlertMessage += ` by User ${ackResponse.acknowledged_by}`;
         }
+        if (ackResponse.unix_millis_acknowledged_time) {
+          const ackSecondsAgo =
+            (new Date().getTime() - ackResponse.unix_millis_acknowledged_time) / 1000;
+          showAlertMessage += ` ${Math.round(ackSecondsAgo)}s ago`;
+        }
+        showAlert('success', showAlertMessage);
+      } else {
+        console.log(`Failed to acknowledge alert ID ${taskAlert.task_id}`);
+        showAlert('error', `Failed to acknowledge alert ID ${taskAlert.task_id}`);
+      }
+    })();
+  };
+
+  if (!taskAlert) {
+    return <></>;
+  }
 
-        return (
-          <AlertDialog
-            key={alert.task_id}
-            onDismiss={dismissAlert}
-            onAcknowledge={acknowledgeAlert}
-            title={alert.title}
-            progress={alert.progress}
-            alertContents={alert.content}
-            backgroundColor={alert.color}
-          />
-        );
-      })}
+  return (
+    <>
+      {taskAlert.acknowledgedBy ? (
+        <AlertDialog
+          key={taskAlert.task_id}
+          onDismiss={removeAlert}
+          acknowledgedBy={taskAlert.acknowledgedBy}
+          onInspect={() => setOpenTaskInspector(true)}
+          title={taskAlert.title}
+          progress={taskAlert.progress}
+          alertContents={taskAlert.content}
+          backgroundColor={taskAlert.color}
+        />
+      ) : (
+        <AlertDialog
+          key={taskAlert.task_id}
+          onDismiss={removeAlert}
+          onAcknowledge={acknowledgeAlert}
+          onInspect={() => setOpenTaskInspector(true)}
+          title={taskAlert.title}
+          progress={taskAlert.progress}
+          alertContents={taskAlert.content}
+          backgroundColor={taskAlert.color}
+        />
+      )}
```

Comment:
Maybe we can simplify it
```suggestion
      <AlertDialog
        key={taskAlert.task_id}
        onDismiss={removeAlert}
        acknowledgedBy={taskAlert.acknowledgedBy}
        onAcknowledge={taskAlert.acknowledgedBy ? undefined : acknowledgeAlert}
        onInspect={() => setOpenTaskInspector(true)}
        title={taskAlert.title}
        progress={taskAlert.progress}
        alertContents={taskAlert.content}
        backgroundColor={taskAlert.color}
      />
```

Reply:
Thanks for catching this, it definitely cleaned up lots, https://github.com/open-rmf/rmf-web/pull/710/commits/781f7a60c4817a44e6b9645fbbd0bb89b98ad2ab